PLC Development/DrivePLC/POUs/FB_AxisDrive.TcPOU

```diff
@@ -0,0 +1,868 @@
+		AxisController.bBusy:=MCGearInMultiMaster.Busy;
+		IF MCGearInMultiMaster.Enable THEN
+			AxisController.bBusy:=TRUE;	//busy does not go immediately high when enable high 
+			bDone:=FALSE;
+		END_IF
+
+		bDone:=MCGearInMultiMaster.InGear;	
+		IF MCGearInMultiMaster.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCGearInMultiMaster.ErrorID;
+		END_IF
+		IF NOT MCGearInMultiMaster.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	eMcCommand.GearOut:		
+		//Gearout - remove all masters from slave
+		FOR iMasterIndex:=1 TO 5 BY 1 DO
+				Limits.coupledFwLimit[GVL.iAxis][iMasterIndex].eLimitType:=eLimitCoupling.None;
+				Limits.coupledBwLimit[GVL.iAxis][iMasterIndex].eLimitType:=eLimitCoupling.None;
+		END_FOR		
+	
+		MCGearOut.Execute:=AxisController.bExecute;
+		
+		AxisController.bBusy:=MCGearOut.Busy;
+		bDone:=MCGearOut.Done;
+		IF MCGearOut.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCGearOut.ErrorID;
+		END_IF
+		IF NOT MCGearOut.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF
+	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////			
+	eMcCommand.Jog:
+		MCJog.JogForward:=AxisController.bJogForward;
+		MCJog.JogBackwards:=AxisController.bJogBackward;
+		
+		bDone:=MCJog.Done;
+		
+		IF MCJog.Error THEN
+			AxisController.bError:=TRUE;
+			AxisController.iErrorID:=MCJog.ErrorID;
+		END_IF
+		IF NOT MCJog.Busy THEN
+			eCommandLatched:=eMcCommand.Idle;
+		END_IF
+	///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////	
+	eMcCommand.HomeStepAbsoluteSwitch:
+		//Issue: If execute not held high the function will not reverse on its own to find the abs switch when hitting a limit
+		//You have to reissue the command. I want to try and keep bExecute high
+		IF (NOT Limits.FwLimit[GVL.iAxis]) OR (NOT Limits.BwLimit[GVL.iAxis]) THEN
+			MCStepAbsoluteSwitch.Execute:=TRUE;
+		ELSE	
+			MCStepAbsoluteSwitch.Execute:=AxisController.bExecute;
+		END_IF
+	
+		MCStepAbsoluteSwitch.Direction:= AxisController.mcHomeDirection;
+		MCStepAbsoluteSwitch.SwitchMode:=AxisController.mcSwitchMode;
+		MCStepAbsoluteSwitch.ReferenceSignal:=AxisController.mcSignalRef;
+		MCStepAbsoluteSwitch.Velocity:=AxisController.fHomeVelocity;
+		MCStepAbsoluteSwitch.Acceleration:=0;
+		MCStepAbsoluteSwitch.Deceleration:=0;
+		MCStepAbsoluteSwitch.Jerk:=0;
+		MCStepAbsoluteSwitch.TorqueLimit:=AxisController.fHomeTorque;
+		MCStepAbsoluteSwitch.SetPosition:=AxisController.fHomePosition;
+		MCStepAbsoluteSwitch.TimeLimit:=AxisController.tHomeTimeLimit;
+		MCStepAbsoluteSwitch.DistanceLimit:=AxisController.fHomeDistanceLimit;
+		MCStepAbsoluteSwitch.PositiveLimitSwitch:= NOT Limits.FwLimit[GVL.iAxis];	//Inverted logic to normal operation
```

Comment:
This comment here is a bit redundant. I can see you're inverting the limit, the comment should either tell me why or not be there.